### PR TITLE
Ability to revoke lease

### PR DESCRIPTION
**Pull request by eyakubovich**
This adds an ability to revoke a lease by deleting it
from etcd. A corresponding API for server case is also added.
Once the lease is revoked, flannel will just acquire a new
one. This has limited use today but is part of #280 to
add reservations. With reservations, there will be a mode
to not acquire a lease if a reservation was not found.

### CODE REVIEW

**Comment by eyakubovich** [comment]
@MohdAhmad @jonboulle PTAL

File: subnet/registry.go [hunk 1]
```diff
@@ -209,6 +211,21 @@ func (esr *etcdSubnetRegistry) watchSubnets(ctx context.Context, network string,
 	return evt, e.Node.ModifiedIndex, err
 }
 
+func (esr *etcdSubnetRegistry) watchSubnet(ctx context.Context, network string, since uint64, sn ip.IP4Net) (Event, uint64, error) {
+	key := path.Join(esr.etcdCfg.Prefix, "subnets", network, sn.StringSep(".", "-"))
```
Comment (dcbw) [inline subnet/registry.go]: Need to swap 'network' and 'subnets' here.

File: subnet/registry.go [hunk 2]
```diff
@@ -127,27 +127,29 @@ func (esr *etcdSubnetRegistry) getSubnets(ctx context.Context, network string) (
...
 		}
+
+		leases = append(leases, *l)
 	}
 
 	return leases, resp.Index, nil
 }
 
+func (esr *etcdSubnetRegistry) getSubnet(ctx context.Context, network string, sn ip.IP4Net) (*Lease, uint64, error) {
+	key := path.Join(esr.etcdCfg.Prefix, network, "subnets", sn.StringSep(".", "-"))
```
Comment (dcbw) [inline subnet/registry.go]: Maybe use "MakeSubnetKey(sn)" here (and everywhere else) instead of StringSep?

**Comment by dcbw** [comment]
Overall LGTM

**Pull request merged**